vars/main.yml

```diff
@@ -7,17 +7,17 @@ prometheus_aggregation_rules:
         expr: "avg(label_replace(netdata_openio_score__average, \"dim2\", \"$1\", \"dimension\", \"(.*?)[_].*\")) \
 by (dim2)"
       - record: "aggr:netdata_web_log_bandwidth:sum"
-        expr: "sum(label_replace(netdata_web_log_bandwidth_kilobits_persec_average, \
-\"chart2\", \"$1\", \"chart\", \".*openio[.](.*)[.].*.log.*\")) by (chart2, instance, dimension)"
+        expr: "sum(label_replace(netdata_web_log_bandwidth_kilobits_persec_average, 'type','$1',\
+        'service','(.*?)-.*')) by (instance, dimension, type)"
```

Comment:
The regex is a bit dangerous as it doesn't handle service types with a `-` (like `event-agent` - even though I don't think we can have that case with this metric). If you want to exclude a character, you should rather use a [negated character class](https://www.regular-expressions.info/charclass.html) instead `[^-]`, but here having a greedy quantifier should be better to handle the aforementioned service types including a `-`, like `(.+)-\d+`.
That being said, instead of doing this `label_replace` 4 times, shouldn't we simply put it in the `relabel_config` of the conf file?

Reply:
(.+)-\d+ makes he assumption that the service id an integer, which is not completely true... Keystone and Event Agent are known to have composite ids (0.0, 0.1 etc...).

Also, I'm not sure we need to tag the type on the non aggregated metrics, as it adds an additional label to all of them. Label replacement is actually quite efficient, when performed during aggregation (as opposed to during queries)

Reply:
Ok. The point still stands for the lazy vs greedy operator.

Reply:
Removed the non-greedy operator

Reply:
:+1: